src/Communication/BTSerialCommunicationManager.cpp

```diff
@@ -0,0 +1,211 @@
+	const char* name = configuration.name.c_str();
+	size_t newsize = strlen(name) + 1;
+	m_wcDeviceName = new WCHAR[newsize];
+	size_t convertedChars = 0;
+	mbstowcs_s(&convertedChars, m_wcDeviceName, newsize, name, _TRUNCATE);
+
+};
+
+void BTSerialCommunicationManager::Connect() {
+	DriverLog("Trying to connect to bluetooth");
+	
+	//We're not yet connected
+	m_isConnected = false;
+
+	//Try to connect
+	if (!getPairedEsp32BtAddress()) //find an ESP32 paired with this machine
+	{
+		DebugDriverLog("Error getting Bluetooth address");
+		return;
+	}
+	if (!startupWindowsSocket()) //initialize windows sockets
+	{
+		DebugDriverLog("Error Initializing windows sockets");
+		return;
+	}
+	if (!connectToEsp32()) //initialize BT windows socket for connecting to ESP32
+	{
+		DebugDriverLog("Error connecting to Bluetooth device");
+		return;
+	}
+	else {
+		//If everything went fine we're connected
+		m_isConnected = true;
+	}
+}
+
+void BTSerialCommunicationManager::BeginListener(const std::function<void(VRCommData_t)>& callback) {
+	DebugDriverLog("Begun listener");
+	m_threadActive = true;
+	m_serialThread = std::thread(&BTSerialCommunicationManager::ListenerThread, this, callback);
+}
+
+void BTSerialCommunicationManager::ListenerThread(const std::function<void(VRCommData_t)>& callback) {
+	//DebugDriverLog("In listener thread");
+	std::this_thread::sleep_for(std::chrono::milliseconds(ARDUINO_WAIT_TIME));
+	PurgeBuffer();
+
+	while (m_threadActive) {
+		std::string receivedString;
+		bool readSuccessful = ReceiveNextPacket(receivedString);
+		
+		if (readSuccessful) {
+			VRCommData_t commData = m_encodingManager->Decode(receivedString);
+
+			callback(commData);
+		}
+		else {
+			DebugDriverLog("Detected that arduino has disconnected! Stopping listener...");
+			//We should probably do more logic for trying to reconnect to the arduino
+			//For now, it should be obvious to people that the arduinos have disconnected
+			m_threadActive = false;
+		}
+
+	}
+}
+
+bool BTSerialCommunicationManager::ReceiveNextPacket(std::string& buff) {
+	//DebugDriverLog("Waiting to recieve a message");
+	char nextChar[1];
+	do {
+		int recievedMessageLength = 1;
+		int recieveResult = recv(m_btClientSocket, nextChar, recievedMessageLength, 0); //if your socket is blocking, this will block until a
+		if (recieveResult < 0)                                                               //a message is recieved. If not, it will return right 
+		{                                                                                    //away
+			continue;
+		}
+		buff += nextChar[0];
+		//printf("Message recieved - \r\n");
+		//DebugDriverLog("Length of buffer: %d, nextchar: %c", buff.length(), nextChar[0]);
+	} while (nextChar[0] != '\n');
+	//DebugDriverLog("Packet received! Length: %d, Packet: %s", buff.length(), buff);
+	
+	return true;
+}
+bool BTSerialCommunicationManager::PurgeBuffer() {
+	return true;//PurgeComm(m_hSerial, PURGE_RXCLEAR | PURGE_TXCLEAR);
+}
```

Comment:
This method doesn't seem to be used and doesn't do anything... I think we can remove it.